misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -549,6 +569,9 @@ private STNode parseValueEnd() {
             case CLOSE_BRACKET_TOKEN:
                 // null marks the end of values
                 return null;
+            case NEWLINE:
+                consume();
+                return parseValueEnd();
```

Comment:
why we need this? what happens to the consumed token in L573?

Reply:
We only use newlines for validation. Ex - There should be a new line after each key value pair. 
But in few cases, in thats different. Here, we are ignoring the new line token passed to parser as that validation is not needed. Ex -  Arrays 
```toml
arr = [
"hello",
"hello1"
,
"hello2"
]
```

Reply:
I see. I guess it should be fine then. Usually syntax tree represents everything including minutiae that user inputs. (useful when formatting, etc)

Anyway since we capture the newline explicitly from the lexer, we need to add NEWLINE as an alternative path for ARRAY_VALUE_END ctx in error handler I think. There could be invalid tokens before the NEWLINE token.
e.g. 
```
[ "abc"  + invalid +
, "ABC" ]
```
@SupunS please have a look.

Reply:
Interesting point.

We designed the error handler to be same as the ballerina error handler, to not to consider newlines. Because newlines are just for validations and parsing could done even without newlines - also simplifies things.

@xlight05 I just noticed that we are not ignoring the newlines in the error-handler though. We should skip the newline tokens or do as @lochana-chathura suggested.

Reply:
+1 for skipping. @xlight05 May be we can create a new issue and address this in a separate PR.

Reply:
Done. #27641